sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,699 @@
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+        self.random_state = random_state
+
+    def _check_params(self):
+        """
+        check model parameters
+        """
+        if self.n_topics <= 0:
+            err_msg = "Invalid 'n_topics' parameter: %r" % self.n_topics
+            raise ValueError(err_msg)
+
+        if self.total_samples <= 0:
+            err_msg = "Invalid 'total_samples' parameter: %r" % self.total_samples
+            raise ValueError(err_msg)
+
+        if self.learning_offset < 0:
+            err_msg = "Invalid 'learning_offset' parameter: %r" % self.learning_offset
+            raise ValueError(err_msg)
+
+        if self.learning_method not in ("batch", "online"):
+            err_msg = "Invalid 'learning_method' parameter: %r" % self.learning_method
+            raise ValueError(err_msg)
+
+    def _init_latent_vars(self, n_features):
+        """
+        Initialize latent variables.
+        """
+        self.rng_ = check_random_state(self.random_state)
+        self.n_iter_ = 1
+        self.n_features = n_features
+
+        if self.doc_topic_prior is None:
+            self.doc_topic_prior_ = 1. / self.n_topics
+        else:
+            self.doc_topic_prior_ = self.doc_topic_prior
+
+        if self.topic_word_prior is None:
+            self.topic_word_prior_ = 1. / self.n_topics
+        else:
+            self.topic_word_prior_ = self.topic_word_prior
+
+        init_gamma = 100.
+        init_var = 1. / init_gamma
+        # In the literature, this is called `lambda`
+        self.components_ = self.rng_.gamma(
+            init_gamma, init_var, (self.n_topics, n_features))
+        # In the literature, this is `E[log(beta)]`
+        self.dirichlet_component_ = _log_dirichlet_expectation(self.components_)
+        # In the literature, this is `exp(E[log(beta)])`
+        self.exp_dirichlet_component_ = np.exp(self.dirichlet_component_)
+
+    def _e_step(self, X, cal_sstats, random_init):
+        """
+        E-step
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Document word matrix.
+
+        cal_sstats : boolean
+            Parameter that indicate whether to calculate sufficient statistics or not.
+            Set `cal_sstats` to `True` when we need to run M-step.
+
+        random_init : boolean
+            Parameter that indicate whether to initialize document topic distribution
+            randomly in the E-step. Set it to `True` in training steps.
+
+        Returns
+        -------
+        (doc_topic_distr, suff_stats) :
+            `doc_topic_distr` is unnormailzed topic distribution for each document.
+            In the literature, this is called `gamma`.
+            `suff_stats` is expected sufficient statistics for the M-step.
+            When `cal_sstats == False`, it will be None.
+
+        """
+
+        # Run e-step in parallel
```

Comment:
This is buggy: it will not implement joblib semantics for n_jobs=-2. Please use sklearn.externals.joblib.cpu_count for that.